### PR TITLE
Customised logger with io.writer

### LINKED ISSUE #2892: Refine logging in integration tests, use customized logger with GinkgoWriter
The library `controller-runtime` requires setting a logger (by `log.SetLogger()`) at the first 30s when the application starts, or it would use the default `NullLogSink`. We should also call it in testing codes.

When we test with ginkgo, ginkgo provides a helpful `GinkgoWriter`, which hides the output as default, only prints it when the test failed. We'd better use it to keep our testing output clean.

So here is the proposal:

- Create another constructor in `pkg/log/zap.go` like `NewDefaultZapLogger` but with the custom `io.Writer`.
- replace `log.SetLogger()` with above logger

Here are several files that need changes:

- api/v1alpha1/suite_test.go
- cmd/chaos-controller-manager/provider/suit_test.go
- controllers/common/suit_test.go
- controllers/schedule/suit_test.go
- pkg/webhook/inject/setup_test.go
- pkg/workflow/controllers/suite_test.go

About using customized with zap, you could create a logger by using `zapcore` directly. Or using `zap.Config` with registering custom sink factory with [RegisterSink](https://pkg.go.dev/go.uber.org/zap#RegisterSink). Reference: https://github.com/uber-go/zap/issues/979

### PR DESCRIPTION
**Pull request by basit9958**
### What problem does this PR solve?

 Close #2892

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [ ] release-2.1
  - [ ] release-2.0

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by basit9958** [comment]
cc @STRRL

**Comment by STRRL** [comment]
Hi @basit9958 , please execute `make check` to fix the CI Check.

File: ui/packages/mui-extends/.storybook/preview.js [hunk 1]
```diff
@@ -6,4 +9,13 @@ export const parameters = {
       date: /Date$/,
     },
   },
+  layout: 'centered',
 }
+
+export const decorators = [
+  (Story) => (
+    <ThemeProvider>
```
[the remark this answers is not in the document]
Comment (STRRL) [reply ui/packages/mui-extends/.storybook/preview.js]: @sonatype-lift ignore

File: ui/packages/mui-extends/.storybook/preview.js [hunk 2]
```diff
@@ -6,4 +9,13 @@ export const parameters = {
...
+      <Story />
+    </ThemeProvider>
```
[the remark this answers is not in the document]
Comment (STRRL) [reply ui/packages/mui-extends/.storybook/preview.js]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (STRRL) [reply ui/packages/mui-extends/.storybook/preview.js]: @sonatype-lift ignore

File: pkg/webhook/inject/setup_test.go [hunk 3]
```diff
@@ -21,6 +21,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/onsi/ginkgo"
```
[the remark this answers is not in the document]
Comment (STRRL) [reply pkg/webhook/inject/setup_test.go:25]: @sonatype-lift ignore

File: cmd/chaos-controller-manager/provider/suit_test.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 	"testing"
 
 	"github.com/go-logr/logr"
+	"github.com/onsi/ginkgo"
```
[the remark this answers is not in the document]
Comment (STRRL) [reply cmd/chaos-controller-manager/provider/suit_test.go:25]: @sonatype-lift ignore

**Comment by basit9958** [comment]
cc @STRRL

File: pkg/log/zap.go [hunk 5]
```diff
@@ -32,3 +35,13 @@ func NewDefaultZapLogger() (logr.Logger, error) {
 	logger := zapr.NewLogger(zapLogger)
 	return logger, nil
 }
+func NewZapLoggerWithWriter(out io.Writer) logr.Logger {
+	bWriter := out
+	config := zap.NewDevelopmentConfig()
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	config.EncoderConfig.FunctionKey = "function"
```
Comment (STRRL) [inline pkg/log/zap.go:42]: on more question, why do we need these configurations, do these changes required? 

rest LGTM!!

**Review by STRRL — APPROVED** [review]
LGTM!

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**